### PR TITLE
[OneExplorer] Ask config file name 

### PR DESCRIPTION
**Pull request by dayo09**
This commit lets user choose a name of new configuration file
on 'create-cfg' command.

ONE-vscode-DCO-1.0-Signed-off-by: Dayoung Lee <dayoung.lee@samsung.com>

### CODE REVIEW

File: src/OneExplorer.ts [hunk 1]
```diff
@@ -140,20 +140,46 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
     this._onDidChangeTreeData.fire();
   }
 
+  /**
+   * Create ONE configuration file for a base model
+   *
+   * If the name is not given by user, it will be named using the base model's file name.
```
Comment (llFreetimell) [inline src/OneExplorer.ts:146]: Not noticed before but I have a question T.T
If recommended file name is already exists, new file would overwrite existing cfg file?

Comment (dayo09) [reply src/OneExplorer.ts:146]: Yeap. We can control it by forbidding overwriting. 😄

Comment (dayo09) [reply src/OneExplorer.ts:146]: I think that is to be handled. User may lose his/her configuration setting by mistake. I will add this.

Comment (llFreetimell) [reply src/OneExplorer.ts]: I am not sure which one would be better, do nothing vs create with alternative recommendation name.
But... let's go as it is :D

**Review by llFreetimell — DISMISSED** [review]
LGTM

File: src/OneExplorer.ts [hunk 2]
```diff
@@ -140,20 +140,47 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
...
+[one-import-${extName}]
+input_path=${modelName}.${extName}
+`);
+
+    vscode.window
+        .showInputBox({
+          title: `Create ONE configuration of '${modelName}.${extName}' :`,
+          placeHolder: `${modelName}.cfg`,
+          value: `${modelName}.cfg`
+        })
```
[the remark this answers is not in the document]
Comment (dayo09) [reply src/OneExplorer.ts:169]: I thought the same so tried to find one, but it was hard to find the proper API. Let me know if you know one :-D. 
We can apply this one first and maybe we can tackle on it later.

File: src/OneExplorer.ts [hunk 3]
```diff
@@ -140,20 +140,47 @@ export class OneTreeDataProvider implements vscode.TreeDataProvider<OneNode> {
...
+    const dirName = path.parse(oneNode.node.path).dir;
+    const modelName = path.parse(oneNode.node.path).name;
+    const extName = path.parse(oneNode.node.path).ext.slice(1);
 
     const encoder = new TextEncoder;
-
     // TODO(dayo) Auto-configure more fields
-    vscode.workspace.fs.writeFile(vscode.Uri.file(`${dirname}/${filename}.cfg`), encoder.encode(`
+    const content = encoder.encode(`
 [one-build]
```
[the remark this answers is not in the document]
Comment (dayo09) [reply src/OneExplorer.ts:158]: Has it been altered? 
Yes I remember that it requries [onecc] on the top of it, but has one-build been ommitted?

I referenced this [[link](https://github.com/Samsung/ONE/blob/master/compiler/one-cmds/how-to-use-one-commands.txt)]

File: src/OneExplorer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (dayo09) [reply src/OneExplorer.ts:169]: FYI, https://stackoverflow.com/questions/70594061/change-an-existing-label-name-in-tree-view-vscode-extension/70604326

**Review by llFreetimell — APPROVED** [review]
LGTM

File: src/OneExplorer.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (jyoungyun) [reply src/OneExplorer.ts:169]: https://github.com/microsoft/vscode/issues/97190

It is not possible to add edit box in tree view.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Comment by dayo09** [comment]
@Samsung/one-vscode_committers  PTAL :-D

**Pull request merged**